wave_vision/src/matcher/brute_force_matcher.cpp

```diff
@@ -53,28 +65,133 @@ void BruteForceMatcher::checkConfiguration(
         throw std::invalid_argument(
           "Norm type is not one of the acceptable values!");
     }
+
+    // Check the value of the ratio_test heuristic
+    if (check_config.ratio_threshold < 0.0 ||
+        check_config.ratio_threshold > 1.0) {
+        throw std::invalid_argument(
+          "ratio_threshold is not an appropriate value!");
+    }
+
+    // Check the value of the threshold distance heuristic
+    if (check_config.distance_threshold < 0) {
+        throw std::invalid_argument("distance_threshold is a negative value!");
+    }
+
+    // Only acceptable values are 1, 2, 4, and 8
+    if (check_config.fm_method < cv::FM_7POINT ||
+        check_config.fm_method > cv::FM_RANSAC) {
+        throw std::invalid_argument("fm_method is not an acceptable value!");
+    }
+
+    if (check_config.fm_method == 3 ||
+        (check_config.fm_method > cv::FM_LMEDS &&
+         check_config.fm_method < cv::FM_RANSAC)) {
+        throw std::invalid_argument("fm_method is not an acceptable value!");
+    }
+}
+
+std::vector<cv::DMatch> BruteForceMatcher::filterMatches(
+  std::vector<cv::DMatch> &matches) const {
+    std::vector<cv::DMatch> filt_matches;
+    float min_distance;
+
+    // Sort matches by distance in increasing order
+    std::sort(matches.begin(), matches.end());
+
+    min_distance = matches.begin()->distance;
+
+    // Keep any match that is less than the rejection heuristic times minimum
+    // distance
+    for (auto &match : matches) {
+        if (match.distance <=
+            this->current_config.distance_threshold * min_distance) {
+            filt_matches.push_back(match);
+        }
```

Comment:
No, so it checks to see if the current match is less than the distance threshold * minimum distance.

If so, it's moved into the new filt_matches vector. The match is never replaced.